
 Make sure grub does not segfault when kernel size is more than 3MB

 When kernel size is more than 3MB stage2 can not load all of kernel, making
load_image in stage2/boot.c unable to identify kerneli, properly identify
boot protocol 2.02 and act due to it. This should only affect userspace command
when kernel command is used, which is not common, yet it may be causing trouble
we are not aware of.


 Onur Küçük <onur@pardus.org.tr>


diff -Nur grub-0.97-old/grub/asmstub.c grub-0.97/grub/asmstub.c
--- grub-0.97-old/grub/asmstub.c	2010-06-04 20:58:12.474543378 +0300
+++ grub-0.97/grub/asmstub.c	2010-06-04 20:58:51.000421436 +0300
@@ -68,7 +68,7 @@
 #include <term.h>
 
 /* Simulated memory sizes. */
-#define EXTENDED_MEMSIZE (3 * 1024 * 1024)	/* 3MB */
+#define EXTENDED_MEMSIZE (5 * 1024 * 1024)	/* 5MB */
 #define CONVENTIONAL_MEMSIZE (640 * 1024)	/* 640kB */
 
 unsigned long install_partition = 0x20000;
